src/app-layout/visual-refresh/drawers.tsx

```diff
@@ -0,0 +1,328 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useRef } from 'react';
+import clsx from 'clsx';
+import customCssProps from '../../internal/generated/custom-css-properties';
+import { IconProps } from '../../icon/interfaces';
+import { InternalButton } from '../../button/internal';
+import { NonCancelableEventHandler } from '../../internal/events';
+import SplitPanel from './split-panel';
+import TriggerButton from './trigger-button';
+import { useAppLayoutInternals } from './context';
+import splitPanelStyles from '../../split-panel/styles.css.js';
+import styles from './styles.css.js';
+import testutilStyles from '../test-classes/styles.css.js';
+
+export interface DrawersProps {
+  activeDrawerId?: string;
+  items: ReadonlyArray<DrawersProps.Drawer>;
+  onChange?: NonCancelableEventHandler<DrawersProps.ChangeDetail>;
+}
+
+namespace DrawersProps {
+  export interface Drawer {
+    ariaLabels?: Labels;
+    content: React.ReactNode;
+    id: string;
+    trigger: Trigger;
+  }
+
+  export interface ChangeDetail {
+    activeDrawerId: string | null;
+  }
+
+  interface Labels {
+    closeButton?: string;
+    content?: string;
+    triggerButton?: string;
+    resizeHandle?: string;
+  }
+  interface Trigger {
+    iconName?: IconProps.Name;
+    iconSvg?: React.ReactNode;
+    iconUrl?: string;
+  }
+}
+
+/**
+ * The Drawers root component is mounted in the AppLayout index file. It will only
+ * render if the drawers are defined, and it will take over the mounting of and
+ * rendering of the Tools and SplitPanel (side position) if they exist. If drawers
+ * do not exist then the Tools and SplitPanel will be handled by the Tools component.
+ */
+export default function Drawers() {
+  const { drawers, hasDrawerViewportOverlay, hasOpenDrawer, isNavigationOpen, navigationHide } =
+    useAppLayoutInternals();
+
+  const isUnfocusable = hasDrawerViewportOverlay && isNavigationOpen && !navigationHide;
+
+  if (!drawers) {
+    return null;
+  }
+
+  return (
+    <div
+      className={clsx(styles['drawers-container'], {
+        [styles['has-open-drawer']]: hasOpenDrawer,
+        [styles.unfocusable]: isUnfocusable,
+      })}
+    >
+      <SplitPanel.Side />
+      <ActiveDrawer />
+      <DesktopTriggers />
+    </div>
+  );
+}
+
+/**
+ * The ActiveDrawer component will render either the drawer content that corresponds
+ * to the activeDrawerId or the Tools content if it exists and isToolsOpen is true.
+ * The aria labels, click handling, and focus handling will be updated dynamically
+ * based on the active drawer or tools content.
+ */
+function ActiveDrawer() {
```

Comment:
I guess, this component is responsible for the open drawer, but unlike #938 there is no resize feature.

What are the plans for this feature?

it is important, because there are implementation suggestions for the classic version (https://github.com/cloudscape-design/components/pull/938/files?w=1#r1162647303), which may also impact how the code is structured here (I am hoping for consistent implementation and features naming across these two versions)